src/accelerate/utils/modeling.py

```diff
@@ -156,11 +156,24 @@ def set_module_tensor_to_device(
             param_cls = type(module._parameters[tensor_name])
             kwargs = module._parameters[tensor_name].__dict__
             if param_cls.__name__ == "Int8Params":
+                # downcast to fp16 if any
+                if new_value.dtype == torch.float32:
+                    new_value = new_value.to(torch.float16)
                 new_value = param_cls(new_value, requires_grad=old_value.requires_grad, **kwargs).to(device)
             else:
                 new_value = param_cls(new_value, requires_grad=old_value.requires_grad).to(device)
             module._parameters[tensor_name] = new_value
 
+            if module.__class__.__name__ == "Linear8bitLt" and getattr(module.weight, "SCB", None) is None:
+                # quantize only if necessary
+                if not getattr(module.weight, "SCB", None) and isinstance(device, int):
+                    if module.bias is not None and module.bias.device.type != "meta":
+                        # if a bias exists, we need to wait until the bias is set on the correct device
+                        module = module.cuda(device)
```

Comment:
Let's use module.to(...)

Reply:
Yes, since we place the model weights on gpu already, we need to "force quantize" the model by calling `cuda()`